physics/module_SGSCloud_RadPre.F90

```diff
@@ -82,97 +94,207 @@ subroutine sgscloud_radpre_run(    &
           do i = 1, im
-            clouds1(i,k) = cldfra_bl(i,k)
-
-            !if( qr(i,k) > 1.0e-7 .OR. qs(i,k) > 1.0e-7.or.qci_conv(i,k)>1.0e-7)THEN
-                !Keep Xu-RandalL clouds fraction - do not overwrite
-            !else
-            !    clouds1(i,k) = cldfra_bl(i,k)
-            !endif
-
-            if (qc(i,k) < 1.e-6 .and. cldfra_bl(i,k)>0.001) then
-              qc(i,k) = qc_bl(i,k)*cldfra_bl(i,k)
-              if (nint(slmsk(i)) == 1) then !land
-                 if(qc(i,k)>1.E-8)clouds3(i,k)=5.4                !eff radius cloud water (microns)
-              else
-                !eff radius cloud water (microns), from Miles et al.
-                if(qc(i,k)>1.E-8)clouds3(i,k)=9.6
-              endif
-              !calculate the liquid water path using additional BL clouds
-              clouds2(i,k) = max(0.0, qc(i,k) * gfac * delp(i,k))
-            endif
-            if (qi(i,k) < 1.e-8 .and. cldfra_bl(i,k)>0.001) then
-              qi(i,k) = qi_bl(i,k)*cldfra_bl(i,k)
-              Tc = T3D(i,k) - 273.15
-              !iwc = qi(i,k)*1.0e6*rho(i,k)
-              if (nint(slmsk(i)) == 1) then !land
-                !eff radius cloud ice (microns), from Mishra et al. (2014, JGR Atmos, fig 6b)
-                if(qi(i,k)>1.E-8)clouds5(i,k)=max(173.45 + 2.14*Tc, 20.)
-              else
-                if(qi(i,k)>1.E-8)clouds5(i,k)=max(173.45 + 2.14*Tc, 20.)
-                !eff radius cloud ice (microns), from Mishra et al. (2014, JGR Atmos, fig 8b)
-                !IF(qi(i,k)>1.E-8)clouds5(i,k)=MAX(139.7 + 1.76*Tc + 13.49*LOG(iwc), 20.)
+            if ( qi(i,k) > 1E-7 .OR. qc(i,k) > 1E-7 ) then
+              es     = min( p3d(i,k),  fpvs( t3d(i,k) ) )  ! fpvs and prsl in pa
+              qsat   = max( QMIN, eps * es / (p3d(i,k) + epsm1*es) )
+              rhgrid = max( 0., min( 0.95, qv(i,k)/qsat ) )
+              h2oliq = qc(i,k) + qi(i,k)                   ! g/kg
+              clwt   = 1.0e-6 * (p3d(i,k)*0.00001)
+
+              if (h2oliq > clwt) then
+                onemrh= max( 1.e-10, 1.0-rhgrid )
+                tem1  = min(max((onemrh*qsat)**0.49,0.0001),1.0)  !jhan                                                          
+                tem1  = 100.0 / tem1
+                value = max( min( tem1*(h2oliq), 50.0 ), 0.0 )
+                tem2  = sqrt( sqrt(rhgrid) )
+
+                clouds1(i,k) = max( tem2*(1.0-exp(-value)), 0.0 )
               endif
-              !calculate the ice water path using additional BL clouds
-              clouds4(i,k) = max(0.0, qi(i,k) * gfac * delp(i,k))
+              !clouds1(i,k)=(1.-exp(-coef_alph*h2oliq/    &
+              !   & ((1.-rhgrid)*qsat*1000.0)**coef_gamm))*(rhgrid**coef_p)
+              !clouds1(i,k)=max(0.0,MIN(1.,clouds1(i,k)))
             endif
-
           enddo
         enddo
-      endif ! do_mynnedmf
 
-      ! add convective clouds
-      if (imfdeepcnv == imfdeepcnv_gf) then
+      else ! kdt > 1 or restart
```

Comment:
this comment assumes that "readers" know that kdt == 1 is the same as flag_init == .true.

Reply:
I can change this to "timestep > 1"

Reply:
Thanks!